Docs: Exclude decorators by default from source

tmeasday opened the pull request:
## What I did

Due to reordering in #21182 the source decorator comes *before* any user decorator in `preview.js`. This means that we commonly trigger the bug in #21649.

Really it was a bug that was fixed by #21182 that the user's decorators *weren't* visible in the source by default given we weren't excluding them.

So it makes more sense to default this to true and people can opt out in cases where they want to show the decorator.

<!-- Thank you for contributing to Storybook! If your PR is related to an issue, provide the number(s) above; if it resolves multiple issues, be sure to break them up (e.g. "closes #1000, closes #1001"). -->

## How to test

Add the following to a React sandbox's `preview.tsx`:

```tsx
  decorators: [
    (S) => <div><S/></div>
  ]
```

Then check the source block of a story:

<img width="418" alt="image" src="https://user-images.githubusercontent.com/132554/226782350-36eefdfe-b9c5-4f1d-bc0b-f4eea73d7ad7.png">


## Checklist

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to make sure your PR is ready to be reviewed. -->

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

<!--

Everybody: Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `main` branch as part of the release process, so you shouldn't need to worry about this. For additional guidance: https://storybook.js.org/docs/react/contribute/how-to-contribute

-->

tmeasday committed:
Exclude decorators by default.

Due to reordering in https://github.com/storybookjs/storybook/pull/21182 the source decorator comes *before* any user decorator in `preview.js`. This means that we commonly trigger the bug in #21649.

Really it makes more sense to default this to true and people can opt out in cases where they want to show the decorator.

tmeasday committed:
Add migration notes

shilman commented on MIGRATION.md:
```diff
@@ -75,7 +75,8 @@
     - [Dropped addon-docs manual configuration](#dropped-addon-docs-manual-configuration)
     - [Autoplay in docs](#autoplay-in-docs)
-    - [Removed STORYBOOK_REACT_CLASSES global](#removed-storybook_react_classes-global)
+    - [Removed STORYBOOK\_REACT\_CLASSES global](#removed-storybook_react_classes-global)
```
Bad formatting

tmeasday replied:
```suggestion
    - [Removed STORYBOOK_REACT_CLASSES global](#removed-storybook_react_classes-global)
```

shilman reviewed (dismissed):
Aside from some markdown issues LGTM. Thanks @tmeasday !

tmeasday commented on MIGRATION.md:
```diff
@@ -31,7 +31,7 @@
   - [7.0 Vite changes](#70-vite-changes)
     - [Vite builder uses Vite config automatically](#vite-builder-uses-vite-config-automatically)
-    - [Vite cache moved to node_modules/.cache/.vite-storybook](#vite-cache-moved-to-node_modulescachevite-storybook)
+    - [Vite cache moved to node\_modules/.cache/.vite-storybook](#vite-cache-moved-to-node_modulescachevite-storybook)
```
```suggestion
    - [Vite cache moved to node_modules/.cache/.vite-storybook](#vite-cache-moved-to-node_modulescachevite-storybook)
```

tmeasday committed:
Undo auto-formatting changes

tmeasday committed:
Merge branch 'next' into tom/21649-change-excludeDecorators-true

tmeasday committed:
Merge branch 'next' into tom/21649-change-excludeDecorators-true

tmeasday committed:
Fix bug with html source decorator

It was returning different rendered code depending on what it was extracting source for.

tmeasday committed:
Apply fix to `web-components` also

shilman committed:
Merge branch 'next' into tom/21649-change-excludeDecorators-true
